Health Check sample using Dependency Injection (#27291)
dotnet/AspNetCore.Docs: aspnetcore/host-and-deploy/health-checks.md

monikerRange: '>= aspnetcore-3.1'
ms.author: riande
ms.custom: mvc
ms.date: 10/19/2022
uid: host-and-deploy/health-checks
---
# Health checks in ASP.NET Core

> By [Glenn Condron](https://github.com/glennc) and [Juergen Gutsch](https://twitter.com/sharpcms)

Reviewer: @JuergenGutsch added you.

Author: @Rick-Anderson Thank you! :-)

For more information, see <xref:fundamentals/middleware/index#branch-the-middleware-pipeline>.

## Dependency Injection and Health Checks

It is possible to use dependency injection to consume an instance of a specific Type inside a Health Check class. This can be useful to inject options or a global configuration to a Health Check.

> Using dependency injection is not a common scenario to configure Health Checks. Usually, each Health Check is quite specific to the actual test and is configured using `IHealthChecksBuilder` extension methods.

The following example shows a sample Health Check that retrieves a configuration object via dependency injection:

:::code language="csharp" source="health-checks/samples/7.x/HealthChecksSample/HealthChecks/SampleHealthCheckWithDI.cs" id="snippet_Class":::

> The used `SampleHealthCheckWithDiConfig` needs to be added to the service container as well as the actual Health Check:

Reviewer: ```suggestion
The `SampleHealthCheckWithDiConfig` and the Health check needs to be added to the service container :
```

Reviewer: @JuergenGutsch  I can fix this if you like. Let me know.

Author: Thank you. It would be cool, if you would fix this.

:::code language="csharp" source="health-checks/samples/7.x/HealthChecksSample/Snippets/Program.cs" id="snippet_MapHealthChecksUsingDependencyInjection":::

> 

Reviewer: This needs to go to line 289 so it shows up in .NET 6 and 7

Author: I just put it on the end to not break any other section. Would you please move it up to the right place? Thank you
